pkg/commands/options/filestuff.go

```diff
@@ -87,11 +87,20 @@ func EnumerateFiles(fo *FilenameOptions) chan string {
 				// the directory, and we're in watch mode, then we set up a watch on the
 				// directory.
 				if fi.IsDir() {
+
 					if path != paths && !fo.Recursive {
 						return filepath.SkipDir
 					}
 					if watcher != nil {
-						watcher.Add(path)
+						if fi.Mode()&os.ModeSymlink == os.ModeSymlink {
+							symlink, err := filepath.EvalSymlinks(path)
+							if err != nil {
+								return err
+							}
+							watcher.Add(symlink)
```

Comment:
This seems to only affect watch mode (`-W`), which is deprecated and will eventually be removed (#412).

Does this also happen to work for non-watch-mode `ko apply -f`?

Reply:
I did something to fix that, hope it'd work 🤦🏻‍♂️